knowledge/store.py

```
@@ -0,0 +1,483 @@
+    except OSError:
+        fallback = Path.home() / ".protoagent" / "knowledge" / "agent.db"
+        fallback.parent.mkdir(parents=True, exist_ok=True)
+        log.info(
+            "[knowledge] %s not writable; using %s instead",
+            p, fallback,
+        )
+        return fallback
+
+
+def _now_iso() -> str:
+    return datetime.now(timezone.utc).isoformat()
```

Ruff (0.15.12)

[warning] Use `datetime.UTC` alias

Convert to `datetime.UTC` alias

(UP017)